### PR TITLE
Feature/implement business logic for civilians

### PR DESCRIPTION
**Pull request by the-17th-fox**
#### Added:
- Three db models
- Two repositories interfaces
- Application services interfaces and their implemetation
- Several viewmodels related to the repositories and services interfaces
- Auth roles, policies and JWT config file.

#### Core
Db models added:
- Identity user db model
- Refresh token db model
- Passport db model

Business logic related to DB:
- Added two repositories interfaces: IPassportsRepository and ITokensRepository
- Applied repository unit of work pattern and added IUnitOfWork interfaces, which will manage Passports and Tokens repository

Application services (interfaces + implementations):
- Added passports service which is used for passports retrieving, updating and creating
- Added users service which handles registration, logining, updating, deleting, [un]blocking the user
- Added tokens service which handled refresh and access token issuing

Configs and other files:
- Added JWT config model, which is used in tokens service for issuing a new access token
- Added auth roles (Default user - for all fresh new users, Admin - for admins)
- Added auth policies (DefaultRights - for all authorized users, Admins - for all admins)
- Region codes - enum which is used as passport region code constraint

Utilities:
- PagedList class for generic items quering from the db.

## Affected Area
Civilians.Core.Auth.
Civilians.Core.Interfaces
Civilians.Core.Misc
Civilians.Core.Models

Civilians.Application.Interfaces
Civilians.Application.Services
Civilians.Application.Utilities
Civilians.Application.ViewModels.*

### CODE REVIEW

File: src/CiviliansAccountsMicroservice/Civilians.Core/Models/Passport.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System.ComponentModel.DataAnnotations;
+
+namespace Civilians.Core.Models
+{
+    [Index(nameof(FirstName), nameof(LastName), nameof(Patronymic))]
```
Comment (KatsiarynaAnashkevich) [inline src/CiviliansAccountsMicroservice/Civilians.Core/Models/Passport.cs:6]: It is better to use FluentApi

File: src/CiviliansAccountsMicroservice/Civilians.Core/Models/RefreshToken.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿using System.ComponentModel.DataAnnotations.Schema;
+using System.ComponentModel.DataAnnotations;
+using Microsoft.EntityFrameworkCore;
+
+namespace Civilians.Core.Models
+{
+    [Index(nameof(UserId), IsUnique = true)]
+    public class RefreshToken
+    {
+        [Key]
```
Comment (KatsiarynaAnashkevich) [inline src/CiviliansAccountsMicroservice/Civilians.Core/Models/RefreshToken.cs:10]: Same

File: src/CiviliansAccountsMicroservice/Civilians.Core/ViewModels/Civilians/LoginViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace Civilians.Core.ViewModels.Civilians
+{
+    /// <summary>
+    /// TODO: Add detailed validation
+    /// </summary>
+
+    public class LoginViewModel
```
Comment (KatsiarynaAnashkevich) [inline src/CiviliansAccountsMicroservice/Civilians.Core/ViewModels/Civilians/LoginViewModel.cs:9]: Do you have model and viewModel on one layer?

Comment (the-17th-fox) [reply src/CiviliansAccountsMicroservice/Civilians.Core/ViewModels/Civilians/LoginViewModel.cs:9]: I choose viewmodels location depending on their place of use. 
If a viewmodel needs to be placed as a parameter in the repository method i'll put it right on the same layer.

Comment (KatsiarynaAnashkevich) [reply src/CiviliansAccountsMicroservice/Civilians.Core/ViewModels/Civilians/LoginViewModel.cs:9]: As a parameter to repository method you should pass entity and do everything with it

Comment (the-17th-fox) [reply src/CiviliansAccountsMicroservice/Civilians.Core/ViewModels/Civilians/LoginViewModel.cs]: Done

**Review by KatsiarynaAnashkevich — APPROVED** [review]

**Pull request merged**